### ISSUE TITLE
Throttle exhaustion issue

### ISSUE THREAD

**Issue by orenmazor** [Shopify/shopify_api#289]
@kevinhughes27 

Is there any way to get the collections of products (like, handles at least) without issueing more API calls? 

If I try to sync the products of a shop, to grab the product collections ends up being at least one extra api call, which speeds up how fast I exhaust my available calls.

https://github.com/Shopify/shopify_api/blob/master/lib/shopify_api/resources/product.rb#L18

I feel as though there's an easier way to get that information that I'm missing.

Alternatively, is there any way to expose the collection on the lineitem of an order? you already expose the vendor, so some product info that isn't related to the order is there already.

/cc @perobertson

**Comment by kevinhughes27** [comment]
@maartenvg @dominiquesr

**Comment by maartenvg** [comment]
@orenmazor There is currently no way to get all products and collection information in the same request. If you care about all collections and all products, you might choose to sync Collects (the join between products and collections) too. By doing `limit: 250` you might be able to get collects that cross multiple collections and products in a single request.

Also, we don't track collection information on the order (line item) level, because a product can belong to many collections and storing that information for each line item is not really that useful.

**Comment by orenmazor** [comment]
@maartenvg yeah that makes a lot of sense. thanks for following up. we'll go with that route for now to unblock us, but I'd like to revisit this at some other point. 250 is too small a limit for a significant chunk of my customers (plus, wireshark shows that the api gem does extra calls around things that we dont expect/want. but thats a separate issue/potential pr)

closing for now
